integratedauth/krb5/krb5.go

```diff
@@ -0,0 +1,268 @@
+//go:build !windows && go1.13
+// +build !windows,go1.13
+
+package krb5
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/jcmturner/gokrb5/v8/client"
+	"github.com/jcmturner/gokrb5/v8/config"
+	"github.com/jcmturner/gokrb5/v8/credentials"
+	"github.com/jcmturner/gokrb5/v8/keytab"
+	"github.com/jcmturner/gokrb5/v8/spnego"
+	"github.com/microsoft/go-mssqldb/integratedauth"
+	"github.com/microsoft/go-mssqldb/msdsn"
+)
+
+var (
+	SetKrbConfig = setupKerbConfig
+	SetKrbKeytab = setupKerbKeytab
+	SetKrbCache  = setupKerbCache
+)
+
+// Kerberos Client State
+type krb5ClientState int
+
+type krb5Auth struct {
+	username   string
+	password   string
+	realm      string
+	serverSPN  string
+	port       uint64
+	krb5Config *config.Config
+	krbKeytab  *keytab.Keytab
+	krbCache   *credentials.CCache
+	krb5Client *client.Client
+	state      krb5ClientState
+}
+
+type Kerberos struct {
+	// Kerberos configuration details
+	Config *config.Config
+
+	// Credential cache
+	Cache *credentials.CCache
+
+	// A Kerberos realm is the domain over which a Kerberos authentication server has the authority
+	// to authenticate a user, host or service.
+	Realm string
+
+	// Kerberos keytab that stores long-term keys for one or more principals
+	Keytab *keytab.Keytab
+}
+
+const (
+	// Initiator states
+	initiatorStart        krb5ClientState = iota
+	initiatorWaitForMutal                 = iota + 2
+	initiatorReady
+)
+
+var (
+	_                integratedauth.IntegratedAuthenticator = (*krb5Auth)(nil)
+	AuthProviderFunc integratedauth.Provider                = integratedauth.ProviderFunc(getAuth)
+)
+
+func init() {
```

Comment:
If my app includes this package, and my connection string doesn't specify a specific authenticator for integrated auth, will it use this authenticator with some default setting ?

IE, I'd like my connection string to be the same on Windows as Linux.

Reply:
Should this init also do:
integratedAuth.DefaultProviderName = "krb5" 
?

Reply:
@shueybubbles this comment by Pete should answer your question https://github.com/microsoft/go-mssqldb/blob/main/integratedauth/auth.go#L23

yes, it will fallback to default provider i.e `ntlm` for unix and `winsspi` for windows if the authenticator is not provided in the connection string.

refer https://github.com/microsoft/go-mssqldb/blob/main/auth_windows.go#L17 and  https://github.com/microsoft/go-mssqldb/blob/main/auth_unix.go#L14

And for your second question, no I dont think so we should do `integratedAuth.DefaultProviderName = "krb5"` as this is meant to be a fallback.

@PeteBassettBet365 this is correct right?